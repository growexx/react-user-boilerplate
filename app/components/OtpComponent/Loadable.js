/**
 *
 * Asynchronously loads the component for OtpComponent
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
