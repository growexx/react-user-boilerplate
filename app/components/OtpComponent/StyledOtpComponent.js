import styled from 'styled-components';
export const StyledOtpComponent = styled.div`
  width: 50%;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  .otpComponentContainerStyle {
    width: 100%;
  }
  .otpComponentInputStyle {
    color: @primary-color;
    border: @border-width-base solid @border-color-base;
    width: 100% !important;
    height: @height-lg;
    border-radius: @border-radius-base+8;
  }
  .otpComponentFocusStyle {
    border: @border-width-base solid @primary-color;
    outline: none;
    border-radius: @border-radius-base+8;
  }
  .otpComponentDisabledStyle {
    background: @disabled-bg;
  }
  .otpComponentErrorStyle {
    border: @border-width-base solid @error-color;
  }
`;
