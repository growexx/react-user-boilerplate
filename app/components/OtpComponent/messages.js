/*
 * OtpComponent Messages
 *
 * This contains all the text for the OtpComponent component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.OtpComponent';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the OtpComponent component!',
  },
});
