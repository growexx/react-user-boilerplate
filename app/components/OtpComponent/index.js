/**
 *
 * OtpComponent
 *
 */
import React from 'react';
import OtpInput from 'react-otp-input';
import PropTypes from 'prop-types';
import { StyledOtpComponent } from './StyledOtpComponent';

class OtpComponent extends React.PureComponent {
  render() {
    return (
      <StyledOtpComponent>
        <OtpInput
          value={this.props.value}
          onChange={this.props.onChange}
          numInputs={this.props.numInputs}
          separator={this.props.separator}
          placeholder={this.props.placeholder}
          // container of inputs
          containerStyle={this.props.containerStyle}
          // each input
          inputStyle={this.props.inputStyle}
          // Style applied or class passed to inputs on focus.
          focusStyle={this.props.focusStyle}
          isDisabled={this.props.isDisabled}
          // Style applied or class passed to each input when disabled.
          disabledStyle={this.props.disabledStyle}
          hasErrored={this.props.hasErrored}
          // Style applied or class passed to each input when errored.
          errorStyle={this.props.errorStyle}
          shouldAutoFocus={this.props.shouldAutoFocus}
          // Restrict input to only numbers.
          isInputNum={this.props.isInputNum}
          // Masks input characters.
          isInputSecure={this.props.isInputSecure}
        />
      </StyledOtpComponent>
    );
  }
}

OtpComponent.propTypes = {
  separator: PropTypes.node,
  numInputs: PropTypes.number,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func,
  placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  isDisabled: PropTypes.bool,
  hasErrored: PropTypes.bool,
  shouldAutoFocus: PropTypes.bool,
  isInputNum: PropTypes.bool,
  isInputSecure: PropTypes.bool,
  containerStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  focusStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  disabledStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  errorStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  inputStyle: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

OtpComponent.defaultProps = {
  separator: <span>-</span>,
  numInputs: 6,
  placeholder: '000000',
  isDisabled: false,
  hasErrored: false,
  shouldAutoFocus: true,
  isInputNum: true,
  isInputSecure: false,
  inputStyle: 'otpComponentInputStyle',
  containerStyle: 'otpComponentContainerStyle',
  focusStyle: 'otpComponentFocusStyle',
  disabledStyle: 'otpComponentDisabledStyle',
  errorStyle: 'otpComponentErrorStyle',
};

export default OtpComponent;
