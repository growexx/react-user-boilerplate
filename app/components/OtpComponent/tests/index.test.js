/**
 *
 * Tests for OtpComponent
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import OtpComponent from '../index';
import configureStore from '../../../configureStore';
let store;
const componentWrapper = Component =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Component />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<OtpComponent />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper(OtpComponent);
    expect(firstChild).toMatchSnapshot();
  });
});
