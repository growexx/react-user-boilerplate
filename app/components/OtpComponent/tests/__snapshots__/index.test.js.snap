// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtpComponent /> Should render and match the snapshot 1`] = `
.c0 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .otpComponentContainerStyle {
  width: 100%;
}

.c0 .otpComponentInputStyle {
  color: #4d186e;
  border: 1px solid #d9d9d9;
  width: 100% !important;
  height: 40px;
  border-radius: 10px;
}

.c0 .otpComponentFocusStyle {
  border: 1px solid #4d186e;
  outline: none;
  border-radius: 10px;
}

.c0 .otpComponentDisabledStyle {
  background: #f5f5f5;
}

.c0 .otpComponentErrorStyle {
  border: 1px solid #ff0000;
}

<div
  class="c0"
>
  <div
    class="otpComponentContainerStyle"
    style="display: flex;"
  >
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Please enter verification code. Digit 1"
        autocomplete="off"
        class="otpComponentInputStyle otpComponentFocusStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
      <span>
        -
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Digit 2"
        autocomplete="off"
        class="otpComponentInputStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
      <span>
        -
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Digit 3"
        autocomplete="off"
        class="otpComponentInputStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
      <span>
        -
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Digit 4"
        autocomplete="off"
        class="otpComponentInputStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
      <span>
        -
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Digit 5"
        autocomplete="off"
        class="otpComponentInputStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
      <span>
        -
      </span>
    </div>
    <div
      style="display: flex; align-items: center;"
    >
      <input
        aria-label="Digit 6"
        autocomplete="off"
        class="otpComponentInputStyle"
        maxlength="1"
        placeholder="0"
        style="width: 1em; text-align: center;"
        type="tel"
        value=""
      />
    </div>
  </div>
</div>
`;
